### PR TITLE
Add SSVersion

### PR DESCRIPTION
**Pull request by errael**
Automatically make the version from pom.xml available to the running program.
Versions can be compared.

### CODE REVIEW

File: swingset/src/main/java/com/nqadmin/swingset/utils/SSVersion.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+		for (int i = 0; i < versionSequence.size(); ++i) {
+			int diff = versionSequence.get(i) - o.versionSequence.get(i);
+			if (diff != 0) {
+				return diff;
+			}
+		}
+		if (isSnapshot == o.isSnapshot) {
+			return 0;
+		}
+		// If this is a snapshot, then it is "smaller" that the other.
```
Comment (bpangburn) [inline swingset/src/main/java/com/nqadmin/swingset/utils/SSVersion.java]: "smaller" than... vs "smaller" that...

File: swingset/src/main/java/com/nqadmin/swingset/utils/SSVersion.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	public static SSVersion get(String sVer) {
+		return parse(sVer);
+	}
+
+	/** Version number components. */
+	private final List<Integer> versionSequence;
+	/** Is the version a snapshot? */
+	private final boolean isSnapshot;
+
+	private SSVersion(List<Integer> _versionSequence, boolean _isSnapshot) {
```
Comment (bpangburn) [inline swingset/src/main/java/com/nqadmin/swingset/utils/SSVersion.java:116]: Know it's not required, but I try to add JavaDoc for protected/private methods just for the developer benefit.

File: swingset-demo/src/main/java/com/nqadmin/swingset/demo/MainClass.java [hunk 3]
```diff
@@ -635,11 +636,15 @@ private static void usage() {
 	 * @param _args - optional command line arguments, which are ignored by this
 	 *              program
 	 */
+	@SuppressWarnings("UseOfSystemOutOrSystemErr")
 	public static void main(final String[] _args) {
```
Comment (bpangburn) [inline swingset-demo/src/main/java/com/nqadmin/swingset/demo/MainClass.java:640]: I was going to ask about logger vs system.x.print, but in a main method I think sending feedback to the terminal makes sense.

File: swingset/pom.xml [hunk 4]
```diff
@@ -173,6 +173,12 @@
 	</dependencies>
 
 	<build>
+		<resources>
```
Comment (bpangburn) [inline swingset/pom.xml:176]: So this modification to the pom is telling it to dynamically modify the swingsetVersion.properties during build and replace ${project.version} with the {version} variable value at the top of the pom? Cool.

**Review by bpangburn — COMMENTED** [review]
Neat!

File: swingset-demo/src/main/java/com/nqadmin/swingset/demo/MainClass.java (hunk 3 above)
Comment (errael) [reply swingset-demo/src/main/java/com/nqadmin/swingset/demo/MainClass.java:640]: Note that outputting the JDK version info has been there for year(s).

File: swingset/pom.xml (hunk 4 above)
Comment (errael) [reply swingset/pom.xml:176]: All the files (both of them) in the directory are filtered/copied to the jar. It didn't seem worth it to create another resource directory.

**Pull request merged**